### PR TITLE
(#262) Build site alert component

### LINKED ISSUE #262: Build: Site Alert Component
### Build alert component based on design specs/requirements documented in this ticket

COVID AND SHUTDOWN ALERTS SHOULD BE IN THIS TICKET

> **ESTIMATE** 3 (this may have been underestimated 5)
Need to complete all tests (PR changes were required) - estimated SP left: 2

## Design Definition of Done
*When all criteria below are met.*

### Baseline Discovery (add any additional info or links)
- [x] Heuristics/Best Practices
- [x] Analytics
*Related links:
- [x] Accessibility
*Related links:

### Design
- [x] Design component
- [x] Conduct Design QA
*Meet with design team to do visual QA for colors, fonts, spacing (padding/margin), etc.
- [x] Get component approved by client

### USWDS Component Equivalent
*Is this a USWDS component?  If so - are there variants? If we are creating something new, provide links to what it is based off of in resources section*

Yes this is a variant on the standard-informational-site-alert: https://designsystem.digital.gov/components/site-alert/

### Styling: 
*Anatomy of a component*
- [ ] Use correct spacing (8px grid)
Size, emphasis, alignment

### Usage
*When to use component and when to use variant(s)*
Ex: Use primary button over secondary

- Use the blue site alert for the COVID content.
- Use the red site alert for government shutdowns.
- When there is more than one link, the site alert becomes a drop down that can be opened and closed to reveal links in a bulleted list (separated by columns for desktop). 
- Users need the opportunity to close a site alert so that it no longer appears during their time on the site. This should be an optional setting in drupal. Ex: gov shutdown site alert would not get the "x".
- Site alerts include the icons from USDWS.

### From USWDS:
**When to use the site alert component**
Critical system notifications. Use a site alert to deliver a static system status update, such as notices about unavailable services or content. Site alerts should appear by default and not in response to an action.
Messages that should be displayed on every page. A site alert makes critical information obvious and findable on every page.

**When to consider something else**
Responding to user actions. Use the alert component for page-level validation messages (like error messages on a form) or to inform a user that a task was completed successfully.
Validating form input. On most forms, especially longer forms, validation messages should appear inline with error messages at the top of the form.

**Usability guidance**
Place prominently. Use a full-width site alert positioned near the top of the page so it’s one of the first things users see.
Avoid stacking multiple site alerts. If you need to convey more than one message, provide a list of links within a single site alert instead of multiple, stacked alerts.
Avoid overwhelming use of color. It may be tempting to make heavy use of bright red or orange to draw attention to an emergency-related site alert. However, these colors can produce strong negative emotional reactions such as fear or panic. Positioning the banner at the top of the screen provides enough weight that such strong colors may not be necessary.

**Accessibility**
Don’t visually hide alert messages and then make them visible when they are needed. Users of older assistive technologies may still be able to perceive hidden alert messages. Fully remove alert messages when they’re not needed.

*Style Guide*
Dos and Donts
Ex: Don't have 1 button in CTA strip
1. Use the COVID alert according to NCI's policy. Do not change icons, text, colors, links, or functionality.
2. Use the Shutdown alert according to NCI's policy. Do not change icons, text, colors, links, or functionality.

### Behaviors
*Summary of state changes and behaviors. AC's are more detailed.*
When the arrow is clicked on, the alert will expand and collapse. When the x is clicked on, the alert will disappear for the remainder of the users' session. The alerts appear beneath the USA banner. When the shutdown banner is added, it appears beneath the USA banner and above other alerts that may be present.

### InVision Mockups: 
*Here goes a URL to the mockup(s) in InVision that include specs*
  * Navigate through different breakpoints: https://publicis.invisionapp.com/project/21811368
  ** The site alert components have been added to the DSM: https://publicis.invisionapp.com/dsm/publicis/nci-ds-new

### Acceptance Criteria for Behaviors 
*If a component changes between breakpoints, or things show/hide, that gets captured here* 

```gherkin
Scenario: User expands and collapses a site alert
  Given user is on a Cancer.gov page (or another NCI site)
    And the site alert has multiple links under it
    And they haven't closed out the site alert previously
  Then the site alert appears and is collapsed
  When user clicks on down arrow in the site alert
  Then the site alert expands and links are exposed
    And the arrow icon changes direction to an upward facing arrow
  When user clicks on up arrow 
  Then site alert collapses 
    And the arrow icon changes direction to a downward facing arrow
```

```gherkin
Scenario: User explores site alert links
  Given user is on a Cancer.gov page (or another NCI site)
    And the site alert has multiple links under it
    And they haven't closed out the site alert previously
  When user clicks on down arrow in the site alert
  Then the site alert expands and links are exposed
    And the arrow icon changes direction to an upward facing arrow
  When user hovers on link in site alert
  Then hover state is applied to link
  When user clicks on an external link in the site alert
  Then the page is loaded
When user clicks on a cgov link 
Then the page loads
    And site alert is collapsed
```

```gherkin
Scenario: User removes site alert
  Given user is on a Cancer.gov page (or another NCI site)
    And they haven't closed out the site alert previously
 When user clicks on the x icon
 Then the site alert disappears
 When user navigates to another Cancer.gov page
 Then the site alert does not appear
```

```gherkin
Scenario: Shutdown site alert does not have multiple links
  Given user is on a Cancer.gov page (or another NCI site)
    And the shutdown site alert is added
    And the site alert does not have multiple links under it
  Then the shutdown site alert appears beneath the USA banner and above any other site alerts
```

---

### Tasks
- [ ] Update Documentation Site (docs)
  - [ ] Create or update [INSERT DOCUMENTATION PAGE OR SECTION ON PAGE HERE] that includes:
    - Developer Documentation
      -  What SASS partial to import
      - (HTML Tab) What is the HTML?
      - (HTML Tab) (for vanilla js) What is the js library to import?
    - **IA Provided** Guidance Documentation  
      - Includes when/how the component should be used
  - [ ] If the component is a global component like the footer, then it needs to be added to the documentation site as well as a real usage.
- [ ] Update NCIDS-CSS
  - site banner with no header 
  - default: banner without collapsible, not shouty
  - variant: `.usa-site-alert--info`
  - variant: `.usa-site-alert--emergency`
  - not real variant: collapsible
    - include new element: `.usa-alert__nci-list`: creates column lists
- [ ] Update NCIDS-JS
  - create nci collapsing site alert
- [ ] Create new ticket for React
- [ ] Create Storybook regression tests

### LINKED ISSUE #344: LivePreview can show bad previews
### Issue description

LivePreview doesn't always show components as expected. See screenshot.  After some investigation, it is caused by an unwanted `<pre>` tag that wraps the code examples.

Possible fix:
- In docs/src/componet/layouts/wrap-root-element.js, uncomment line 17: `pre: (props) => props.children`
- Make sure this works as expected in all documentation pages.

> **ESTIMATE** TBD

### Additional details / screenshot
- Screen Shot 2022-02-23 at 3 41 28 PM
- Screen Shot 2022-02-24 at 9 08 04 AM

### Related Tickets
* #332

### PR DESCRIPTION
**Pull request by olitharp-nci**
Closes #262
Closes #344

https://designsystem-dev.cancer.gov/pr-319/components/site-alert

### CODE REVIEW

File: docs/content/design-tokens/state-tokens.mdx [hunk 1]
```diff
@@ -45,7 +45,7 @@ not every family needs to include a color at each grade. Some grades may be set
 | <ColorChip color={"#e6e6e6"} /> | `'disabled-light'`  | `'gray-10'`      | `$theme-color-disabled-light`  | `#e6e6e6` |
 | <ColorChip color={"#c9c9c9"} /> | `'disabled'`        | `'gray-20'`      | `$theme-color-disabled`        | `#c9c9c9` |
 | <ColorChip color={"#adadad"} /> | `'disabled-dark'`   | `'gray-30'`      | `$theme-color-disabled-dark`   | `#adadad` |
-| <ColorChip color={"#9c3d10"} /> | `'emergency'`       | `'red-warm-60v'` | `$theme-color-emergency`       | `#9c3d10` |
+| <ColorChip color={"#bb0e3d"} /> | `'emergency'`       | `'cranberry-60'` | `$theme-color-emergency`       | `#bb0e3d` |
```
Comment (bryanpizzillo) [inline docs/content/design-tokens/state-tokens.mdx:48]: Does emergency-dark change as well? I would assume these two are related.

File: packages/ncids-css/scss/base/settings/_settings-color.scss [hunk 2]
```diff
@@ -114,7 +114,7 @@ $theme-color-disabled: 'gray-20' !default;
 $theme-color-disabled-dark: 'gray-30' !default;
 
 // Emergency colors
-$theme-color-emergency: 'red-warm-60v' !default;
+$theme-color-emergency: 'cranberry-60' !default;
 $theme-color-emergency-dark: 'red-warm-80' !default;
```
Comment (bryanpizzillo) [inline packages/ncids-css/scss/base/settings/_settings-color.scss:118]: See my above comment about emergency dark.

File: packages/ncids-css/scss/components/usa-site-alert/_all.scss [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// USWDS Component
+@import 'uswds/dist/scss/components/site-alert';
+
+// Dependencies
+@import 'uswds/dist/scss/components/alerts';
+
+// NCI Components
+@import "internal/nci-emergency";
```
Comment (bryanpizzillo) [inline packages/ncids-css/scss/components/usa-site-alert/_all.scss:8]: This should just be
```
@import "./nci-emergency"
@import "./nci-info"
```

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
Some initial notes.

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
Can you please add a link on the documentation site homepage to the site alert?

**Comment by mworrest** [comment]
I am seeing purple for the link color on the blue alert. It should be blue. Is there going to be the open/close functionality or is that in another ticket?

Screen Shot 2022-02-24 at 6 05 02 PM

**Comment by olitharp-nci** [comment]
> I am seeing purple for the link color on the blue alert. It should be blue. Is there going to be the open/close functionality or is that in another ticket?
> 
> Screen Shot 2022-02-24 at 6 05 02 PM

A purple link usually means it's a visited href. I'll take a look.

**Comment by olitharp-nci** [comment]
@mworrest, it should be blue now as expected, let me know if you still see purple 

Screen Shot 2022-02-25 at 8 42 58 AM

**Comment by meglake** [comment]
Michelle creating design ticket (will be a spike) to look into changing the link color across the site to avoid contrast issue noted in the PR and determine what happens if we make the dark be the default for link color. Putting in design backlog, potentially for sprint 25.

**Comment by mworrest** [comment]
Created spike ticket: https://github.com/NCIOCPL/ncids/issues/348

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
TO FIX:
* Add the component to scss/components/all
* Fix gatsby-browser to import all, and update docs on how to use uswds.
* Also there are nits having to do with how we document the TS types and not using `@typedef` but doing the comments over the elements

FOR FUTURE:
2 issues that we can track as separate items:
1. BUG - The NCICloseButton for the SiteAlert uses the same Cookie name.
2. REQ ISSUE - The Expand/Collapse should also use a cookie

**Comment by olitharp-nci** [comment]
https://github.com/NCIOCPL/ncids/issues/349
https://github.com/NCIOCPL/ncids/issues/350

File: docs/content/components/site-alert.mdx [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+### Usability guidance
+
+- **Place prominently.** Use a full-width site alert positioned near the top of the page so it’s one of the first things users see.
+- **Avoid stacking multiple site alerts.** If you need to convey more than one message, provide a list of links within a single site alert instead of multiple, stacked alerts.
+- **Avoid overwhelming use of color.** It may be tempting to make heavy use of bright red or orange to draw attention to an emergency-related site alert. However, these colors can produce strong negative emotional reactions such as fear or panic. Positioning the banner at the top of the screen provides enough weight that such strong colors may not be necessary.
+
+### Accessibility
+
+- **Don’t visually hide alert messages and then make them visible when they are needed.** Users of older assistive technologies may still be able to perceive hidden alert messages. Fully remove alert messages when they’re not needed.
+- `aria-label` must be unique and descriptive to what the alert is so that screen readers
```
Comment (arcepaul) [inline docs/content/components/site-alert.mdx:38]: unfinished thought

File: docs/content/components/site-alert.mdx [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+					</li>
+				</ul>
+			</div>
+		</div>
+	</div>
+</section>
+```
+
+### Dismissible site alert
+
```
Comment (arcepaul) [inline docs/content/components/site-alert.mdx:192]: I don't see the cross for the dismissible

**Review by bryanpizzillo — APPROVED** [review]
This all looks good to me.

**Review by arcepaul — APPROVED** [review]

**Review by mworrest — APPROVED** [review]

**Pull request merged**